reactor-core/src/test/java/reactor/core/publisher/AbstractOnDiscardShouldNotLeakTest.java

```diff
@@ -0,0 +1,443 @@
+package reactor.core.publisher;
+
+import org.assertj.core.api.Assertions;
+import org.assertj.core.api.Assumptions;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.reactivestreams.Publisher;
+import reactor.core.Fuseable;
+import reactor.core.Scannable;
+import reactor.core.scheduler.Scheduler;
+import reactor.core.scheduler.Schedulers;
+import reactor.test.publisher.TestPublisher;
+import reactor.test.subscriber.AssertSubscriber;
+import reactor.test.util.RaceTestUtils;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+@RunWith(Parameterized.class)
+public abstract class AbstractOnDiscardShouldNotLeakTest {
```

Comment:
Have you considered adding this to `BaseOperatorTest` instead?

Reply:
Yeah. I don't know how to use it properly, thus created an abstract class for simplicity at the first stage, but ready to optimize and make it extending `BaseOperatorTest`